### PR TITLE
create 0016-internal-like-system

### PR DESCRIPTION
**Pull request by RubenSmn**
ADR on an Intern Like system

Did some brainstorming with @dadiorchen about the current way of giving a like, resulting in this document

Shortcut to goto the markdown page: https://github.com/RubenSmn/treetracker-decisions/blob/main/0016-intern-like-system.md

### CODE REVIEW

**Comment by dadiorchen** [comment]
Some background, on the web map we are going to add a `like` button on trees, currently, we are using FB like button, but it has some limitation, and we don't have the data of that, for example: top liked tree on treetracker.
So another idea is to build our own like button. and also to consider if it is also useful for admin panel, and android/ios treetracker app? 
Please give some input @sebastiangaertner @nmcharlton @Elforama

File: 0016-intern-like-system.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Intern Like system
```
Comment (nmcharlton) [inline 0016-intern-like-system.md:1]: Should be "internal" rather than "intern".

File: 0016-intern-like-system.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+Option #1
+* Users do not need an social media account, for example Facebook
+* The like data becomes our own
+* Implementation of a custom UI Like button would be possible, currently we have a standard Facebook Like button
+
+### Negative Consequences <!-- optional -->
+
+Option #1
+* Risk of "cheating", not being able to identify the user
+
```
Comment (nmcharlton) [inline 0016-intern-like-system.md:43]: If we don't link "likes" to a user account, the user experience may be confusing. A user returning to a previously "liked" tree may see no record that they liked it, causing them to do so again.

File: 0016-intern-like-system.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+Another issue is that we are not able to create a custom Like button in the Greenstand style, https://github.com/Greenstand/treetracker-web-map-client/pull/395
+
+## Decision Drivers <!-- optional -->
+
+* {driver 1, e.g., a force, facing concern, …}
+
+## Considered Options
+
+* Option #1 - Intern Like system, to save likes on trees/entities directly
+* Option #2 - Your suggestions are welcome
```
Comment (nmcharlton) [inline 0016-intern-like-system.md:24]: Option #2 – Internal Like system linked with Keycloak user authentication system

Comment (nmcharlton) [reply 0016-intern-like-system.md:24]: The need to log in before liking/saving an entity is a very common pattern, so would be expected by most users. The "like" button is a great way to ease users into creating an account/wallet with Greenstand.

File: 0016-intern-like-system.md (hunk 2 above)
Comment (dadiorchen) [reply 0016-intern-like-system.md:43]: I'm also worried about this, but there might be some trade-off way to identify the user, by collecting some fingerprint based on browser agent or so? or use cookie, locale storage

File: 0016-intern-like-system.md (hunk 3 above)
Comment (dadiorchen) [reply 0016-intern-like-system.md:24]: I also prefer the keycloak/SSO solution

**Comment by dadiorchen** [comment]
@RubenSmn should we change the status of this PR to something indicating that we have made the decision? I think now we all agree with option 2, right?

```
Status: proposed
Input: Welcome from anyone!
Deciders: {list everyone involved in the decision using github handle}
Date: 2022-09-09
```

**Comment by sebastiangaertner** [comment]
Agree

**Pull request merged**